src/Centreon/Application/Controller/DowntimeController.php

```diff
@@ -0,0 +1,475 @@
+        $withService = $postData['with_services'] ?? false;
+        $this->monitoringService->filterByContact($contact);
+        $host = $this->monitoringService->findOneHost($hostId);
+
+        if ($host === null) {
+            return View::create(null, Response::HTTP_NOT_FOUND, []);
+        }
+
+        $this->downtimeService->filterByContact($contact);
+
+        /**
+         * @var Downtime $downtime
+         */
+        $downtime = $serializer->deserialize(
+            (string)$request->getContent(),
+            Downtime::class,
+            'json'
+        );
+
+        $this->downtimeService->addHostDowntime($downtime, $host);
+
+        if ($withService) {
+            $services = $this->monitoringService->findServicesByHost($hostId);
+            foreach ($services as $service) {
+                $service->setHost($host);
+            }
+            $this->downtimeService->addServicesDowntime($downtime, $services);
+        }
+        return $this->view();
+
+    }
+
+    /**
+     *  Entry point to add a service downtime
+     *
+     * @IsGranted("ROLE_API_REALTIME", message="You are not authorized to access this resource")
+     * @Rest\Post(
+     *     "/monitoring/hosts/{hostId}/services/{serviceId}/downtimes",
+     *     condition="request.attributes.get('version.is_beta') == true",
+     *     name="monitoring.downtime.addServiceDowntime")
+     * @param Request $request
+     * @param JsonValidatorInterface $jsonValidator
+     * @param SerializerInterface $serializer
+     * @param int $hostId Host id linked to the service
+     * @param int $serviceId Service id for which we want to add a downtime
+     * @param string $version
+     * @return View
+     * @throws \Exception
+     */
+    public function addServiceDowntime(
+        Request $request,
+        JsonValidatorInterface $jsonValidator,
+        SerializerInterface $serializer,
+        int $hostId,
+        int $serviceId,
+        string $version
+    ): View {
+        /**
+         * @var Contact $contact
+         */
+        $contact = $this->getUser();
+        if ($contact === null || (!$contact->isAdmin() && !$contact->hasRole(Contact::ROLE_ADD_SERVICE_DOWNTIME))) {
+            return $this->view(null, Response::HTTP_UNAUTHORIZED);
+        }
+
+        $errors = $jsonValidator
+            ->forVersion($version)
+            ->validate((string) $request->getContent(), 'add_downtime');
+
+        if ($errors->count() > 0) {
+            throw new ValidationFailedException($errors);
+        } else {
+            /**
+             * @var Downtime $downtime
+             */
+            $downtime = $serializer->deserialize(
+                (string) $request->getContent(),
+                Downtime::class,
+                'json'
+            );
+            $this->monitoringService->filterByContact($contact);
+
+            $host = $this->monitoringService->findOneHost($hostId);
```

Comment:
Do you need to make this check ?
Cause you already check if service (and/or host) exists in the next lines

Reply:
You're right